### DISCUSSION TITLE
System include extraction: not allowed driver /mypath/aarch64-linux-gnu-g++

### DISCUSSION THREAD

**Discussion by Piratezoe** [clangd/clangd#1507] [Q&A]
When I try to cross-compile on my WSL-2 ubuntu 18.04 with the toolchain aarch64-linux-gnu-g++
cland seems can't recognized the toolchain to extra the std related content.

Is there anyone who have met this problem?

Thanks Very Much if you can give some advice. :)

Below are my verbose log of clangd

> I[20:55:03.343] clangd version 15.0.6 (https://github.com/llvm/llvm-project 088f33605d8a61ff519c580a71b1dd57d16a03f8)
> I[20:55:03.343] Features: linux+grpc
> I[20:55:03.343] PID: 14666
> I[20:55:03.343] Working directory: /home/pirate/work/recipes-vvv-sop2
> I[20:55:03.343] argv[0]: /home/pirate/.config/coc/extensions/coc-clangd-data/install/15.0.6/clangd_15.0.6/bin/clangd
> I[20:55:03.343] argv[1]: -log=verbose
> I[20:55:03.343] argv[2]: --query-driver=/mnt/c/Work/09.CompileNeed/37W_MQB/toolchain/aarch64-linux-gnu/1/workspace/bin/aarch64-linux-gnu-gcc
> I[20:55:03.343] argv[3]: --query-driver=/mnt/c/Work/09.CompileNeed/37W_MQB/toolchain/aarch64-linux-gnu/1/workspace/bin/aarch64-linux-gnu-g++
> V[20:55:03.343] User config file is /home/pirate/.config/clangd/config.yaml

**omit the Parsing process of .clangd file ...**

> I[20:55:03.647] Loaded compilation database from /home/pirate/work/recipes-vvv-sop2/dev/build/phone/bt-stafk/tsd-blueaaa-hcc-dev/1/workspace/compile_commands.json
> V[20:55:03.648] Broadcasting compilation database from /home/pirate/work/recipes-vvv-sop2/dev/build/phone/bt-stafk/tsd-blueaaa-hcc-dev/1/workspace
> V[20:55:03.648] **System include extraction: not allowed driver** /home/pirate/work/recipes-vvv-sop2/dev/dist/toolchain/aarch64-linux-gnu/1/workspace/bin/aarch64-linux-gnu-g++
> I[20:55:03.649] ASTWorker building file /home/pirate/work/recipes-vvv-sop2/dev/src/phone/bt-stafk/tsd-blueaaa-hcc/1/workspace/tsd.blueaaa.hcc/src/tsd/blueaaa/manage/local/xxx.cpp version 1 with command 
> [/home/pirate/work/recipes-vvv-sop2/dev/build/phone/bt-stafk/tsd-blueaaa-hcc-dev/1/workspace/tsd.blueaaa.hcc]
> /home/pirate/work/recipes-vvv-sop2/dev/dist/toolchain/aarch64-linux-gnu/1/workspace/bin/aarch64-linux-gnu-g++ --target=aarch64-linux-gnu --driver-mode=g++ --sysroot=/home/pirate/work/recipes-vvv-sop2/dev/dist/toolchain/aarch64-linux-gnu/1/workspace/gcc-linaro-7.3.1-2018.05-x86_64_aarch64-linux-gnu/aarch64-linux-gnu/libc -DCOMMLIB_OS=OS_TSD -DEXT_LIBGIF_DLLEXPORT= -DHAVE_CONFIG_H -DLIB_EXPORT= -DTARGET -DTARGET_CPU=1 -DTARGET_CPU_ARM64=1 -DTARGET_CPU_ARM64_V8=1 -DTARGET_OS=1 -DTARGET_OS_POSIX=1 -DTARGET_OS_POSIX_LINUX=1 -DTARGET_TYPE=1 -DTARGET_TYPE_EMBEDDED=1 -DTSD_blueaaa_API_DLLEXPORT= -DTSD_blueaaa_hcc_DLLEXPORT= -DTSD_COMMON_API_VERSION=210 -DTSD_COMMON_BUILD_SHARED_LIB=0 -DTSD_COMMON_CHECKSUMS_API_VERSION=1 -DTSD_COMMON_CHECKSUMS_DLLEXPORT= -DTSD_COMMON_CONFIG_CONFIGMGR_NOFS=0 -DTSD_COMMON_DISABLE_STACKTRACE_COLLECT_IN_EXCEPTIONS=0 "-DTSD_COMMON_DLLEXPORT=__attribute__((visibility(\"default\")))" -DTSD_COMMON_LOGGING_INI_ALT_PATH=/etc/logging/logging.ini "-DTSD_COMMON_LOGGING_INI_ALT_PATH_STRING=\"/etc/logging/logging.ini\"" -DTSD_COMMON_LOGGING_INI_PATH=/var/lib/logging/logging.ini "-DTSD_COMMON_LOGGING_INI_PATH_STRING=\"/var/lib/logging/logging.ini\"" -DTSD_COMMON_LOGGING_USE_JOURNAL=0 -DTSD_COMMON_STACKTRACE_PERF_CRITICAL_EXCEPTIONS=0 -DTSD_COMMON_SYSTEM_PRIORITIES=1 -DTSD_COMMON_UTILS_SCOPE_PROFILER=1 -DTSD_I18N_DLLEXPORT= -DTSD_MEDIA_PICSERVER_DLLEXPORT= -DTSD_MEDIA_PICSERVER_VERSION=21300 -DTSD_PARSER_VOBJECT_API_DLLEXPORT= -DTSD_PARSER_VOBJECT_DLLEXPORT= -DTSD_PHONE_COMMON_DLLEXPORT= -DTSD_THREADMESSAGES_DLLEXPORT= -I/home/pirate/work/recipes-vvv-sop2/dev/src/phone/bt-stafk/tsd-blueaaa-hcc/1/workspace/tsd.blueaaa.hcc/src -I/home/pirate/work/recipes-vvv-sop2/dev/src/phone/bt-stafk/tsd-blueaaa-hcc/1/workspace/tsd.blueaaa.hcc/public -I/home/pirate/work/recipes-vvv-sop2/dev/build/phone/bt-stafk/tsd-blueaaa-hcc-dev/1/workspace/tsd.blueaaa.hcc/src -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/phone/bt-stafk/tsd-blueaaa-api-dev/1/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/system/tsd-common-dev/2/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/phone/tsd-parser-vobject-api-dev/1/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/phone/tsd-phone-common-dev/1/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/system/tsd-threadmessages-dev/1/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/phone/tsd-parser-vobject-dev/1/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/hmi/tsd-i18n-dev/1/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/tip/ext-icu-dev/2/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/media/pictureserver/tsd-media-picserver-dev/1/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/libs/ext-libpng-dev/1/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/libs/ext-zlib-dev/3/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/libs/ext-libjpeg-dev/1/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/libs/ext-libgif-dev/1/workspace/usr/include -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/phone/bt-stafk/ext-hcc-dev/1/workspace/usr/local/include/hcc_api -isystem /home/pirate/work/recipes-vvv-sop2/dev/dist/phone/bt-stafk/ext-hcc-dev/1/workspace/usr/local/include/config -pipe -feliminate-unused-debug-types -march=armv8-a -ffunction-sections -fdata-sections -std=c++11 -Wno-error=deprecated -Wall -Wextra -Werror -Wno-error=deprecated-declarations -g -Os -fPIC -fvisibility=hidden -Wall -Wconversion -Wextra -Wunused -Wempty-body -o CMakeFiles/tsd.blueaaa.hcc.static.pi.dir/src/tsd/blueaaa/manage/local/DeviceSettingRequest.cpp.o -c -resource-dir=/home/pirate/.config/coc/extensions/coc-clangd-data/install/15.0.6/clangd_15.0.6/lib/clang/15.0.6 -- /home/pirate/work/recipes-vvv-sop2/dev/src/phone/bt-stafk/tsd-blueaaa-hcc/1/workspace/tsd.blueaaa.hcc/src/tsd/blueaaa/manage/local/DeviceSettingRequest.cpp
> I[20:55:03.657] --> window/workDoneProgress/create(0)
> V[20:55:03.657] >>> {"id":0,"jsonrpc":"2.0","method":"window/workDoneProgress/create","params":{"token":"backgroundIndexProgress"}}

**Comment by HighCommander4** [comment]
Answered in https://github.com/clangd/clangd/issues/1508. There's no need to file in duplicate.

**Comment by Piratezoe** [comment]
Resolved in https://github.com/clangd/clangd/issues/1508#issuecomment-1435925595
